definitions/scenarios/backup.rb

```diff
@@ -0,0 +1,190 @@
+        Procedures::Backup::Online::CandlepinDB => :backup_dir,
+        Procedures::Backup::Online::ForemanDB => :backup_dir,
+        Procedures::Backup::Offline::CandlepinDB => :backup_dir,
+        Procedures::Backup::Offline::ForemanDB => :backup_dir,
+        Procedures::Backup::Offline::Mongo => :backup_dir,
+        Procedures::Backup::Snapshot::LogicalVolumeConfirmation => :backup_dir,
+        Procedures::Backup::Snapshot::MountCandlepinDB => :backup_dir,
+        Procedures::Backup::Snapshot::MountForemanDB => :backup_dir,
+      )
+      context.map(:preserve_dir,
+        Checks::Backup::DirectoryReady => :preserve_dir,
+        Procedures::Backup::PrepareDirectory => :preserve_dir,
+      )
+      context.map(:incremental_dir,
+        Procedures::Backup::PrepareDirectory => :incremental_dir,
+      )
+      context.map(:proxy_features,
+        Procedures::Backup::ConfigFiles => :proxy_features,
+      )
+      context.map(:snapshot_mount_dir,
+        Procedures::Backup::Snapshot::PrepareMount => :mount_dir,
+        Procedures::Backup::Snapshot::MountMongo => :mount_dir,
+        Procedures::Backup::Snapshot::MountPulp => :mount_dir,
+        Procedures::Backup::Snapshot::CleanMount => :mount_dir,
+        Procedures::Backup::Snapshot::MountCandlepinDB => :mount_dir,
+        Procedures::Backup::Snapshot::MountForemanDB => :mount_dir,
+        Procedures::Backup::Offline::Mongo => :mount_dir,
+        Procedures::Backup::Pulp => :mount_dir,
+        Procedures::Backup::Offline::CandlepinDB => :mount_dir,
+        Procedures::Backup::Offline::ForemanDB => :mount_dir,
+      )
+      context.map(:snapshot_block_size,
+        Procedures::Backup::Snapshot::MountMongo => :block_size,
+        Procedures::Backup::Snapshot::MountPulp => :block_size,
+        Procedures::Backup::Snapshot::MountForemanDB => :block_size,
+      )
+      context.map(:skip_pulp_content,
+        Procedures::Backup::Pulp => :skip,
+        Procedures::Backup::Snapshot::LogicalVolumeConfirmation => :skip_pulp,
+        Procedures::Backup::Snapshot::MountPulp => :skip,
+      )
+      context.map(:tar_volume_size,
+        Procedures::Backup::Pulp => :tar_volume_size,
+        Procedures::Backup::Offline::Mongo => :tar_volume_size,
+        Procedures::Backup::Offline::CandlepinDB => :tar_volume_size,
+        Procedures::Backup::Offline::ForemanDB => :tar_volume_size,
+      )
+    end
+
+    private
+
+    def strategy
+      context.get(:strategy)
+    end
+
+    def add_online_backup_steps
+      add_step_with_context(Procedures::Backup::Pulp, :ensure_unchanged => true)
+      add_steps_with_context(
+        Procedures::Backup::Online::Mongo,
+        Procedures::Backup::Online::PgGlobalObjects,
+        Procedures::Backup::Online::CandlepinDB,
+        Procedures::Backup::Online::ForemanDB,
+      )
+    end
+
+    def logical_backup?
+      context.get(:logical_backup)
+    end
+
+    def online_backup?
+      !!(strategy == :online)
```

Comment:
Here, can we use `strategy == :online` directly instead of `!!`mark as it is returning a Boolean value?